### PR TITLE
Add BlackBoxOptim dispatch to sciml_train

### PR DESCRIPTION
**Pull request by abhigupta768**

### CODE REVIEW

File: src/train.jl [hunk 1]
```diff
@@ -308,3 +308,26 @@ function sciml_train(loss, θ, opt::Optim.AbstractConstrainedOptimizer,
...
                          f_calls_limit = maxiters))
 end
+
+
+function sciml_train(loss, opt::Symbol = :adaptive_de_rand_1_bin, data = DEFAULT_DATA;lower_bounds, upper_bounds,
+                      cb = (args...) -> (false), maxiters = get_maxiters(data), kwargs...)
+  local x, cur, state
+  cur,state = iterate(data)
+
+  function _cb(trace)
```
Comment (Vaibhavdixit02) [inline src/train.jl]: The callback is not being passed here?

**Comment by Vaibhavdixit02** [comment]
I think you should add Evolutionary.jl dispatch to the same PR. Let's just close the issue in a go 🙂

**Comment by abhigupta768** [comment]
I took a look at Evolutionary.jl, they don't have an uniform way of calling optimize with different algorithms. So we would need dispatch to each method independently, not sure if that is desirable?

**Comment by Vaibhavdixit02** [comment]
@ChrisRackauckas what do you think?

**Comment by ChrisRackauckas** [comment]
Dispatching on symbol isn't too safe... I would prefer it to be like `BBO(:method)` or something like that.

File: src/train.jl [hunk 2]
```diff
@@ -308,3 +308,17 @@ function sciml_train(loss, θ, opt::Optim.AbstractConstrainedOptimizer,
...
+                      maxiters = get_maxiters(data), kwargs...)
+  local x, cur, state
+  cur,state = iterate(data)
+
+  _loss = function (θ)
+    x = loss(θ,cur...)
+    first(x)
+  end
+
+  BlackBoxOptim.bboptimize(_loss;Method = opt, SearchRange = [(lower_bounds[i], upper_bounds[i]) for i in 1:length(lower_bounds)], kwargs...)
```
Comment (ChrisDeGrendele) [inline src/train.jl]: @ChrisRackauckas This should then be converted to an Optim.jl optimum solution? To keep `sciml_train` consistent

Comment (ChrisRackauckas) [reply src/train.jl]: yes

File: src/train.jl [hunk 3]
```diff
@@ -308,3 +308,48 @@ function sciml_train(loss, θ, opt::Optim.AbstractConstrainedOptimizer,
            Optim.Options(extended_trace=true,callback = _cb,
                          f_calls_limit = maxiters))
 end
+
+struct BBO
+  method::Symbol = :adaptive_de_rand_1_bin
```
Comment (ChrisRackauckas) [inline src/train.jl]: I don't think you can set defaults like that.

File: src/train.jl [hunk 4]
```diff
@@ -309,8 +309,11 @@ function sciml_train(loss, θ, opt::Optim.AbstractConstrainedOptimizer,
                          f_calls_limit = maxiters))
 end
 
+struct BBO
+  method::Symbol = :adaptive_de_rand_1_bin         
+end
```
Comment (ChrisRackauckas) [inline src/train.jl]:
```suggestion
  method::Symbol         
end
BBO() = BBO(:adaptive_de_rand_1_bin)

```

**Comment by Vaibhavdixit02** [comment]
@ChrisRackauckas I can't see the Travis run for this anymore, Travis needs some update?

**Pull request merged**